src/core/Metavariable.ml

```diff
@@ -156,6 +156,20 @@ let mvalue_of_any = function
   | Anys _ ->
       None
 
+let location_aware_equal_mvalue mval1 mval2 =
+  let ranges_equal =
+    let any1, any2 = (mvalue_to_any mval1, mvalue_to_any mval2) in
+    let range1, range2 =
+      ( AST_generic_helpers.range_of_any_opt any1,
+        AST_generic_helpers.range_of_any_opt any2 )
+    in
+    match (range1, range2) with
+    | Some range1, Some range2 ->
+        [%eq: Tok.location * Tok.location] range1 range2
```

Comment:
note a fan of those [%eq] stuff.

Reply:
Really? I actually like it a bit for generating boilerplate equality functions on typical type constructors, like tuples. That way I don't have to pull the values out myself.

But, I don't mind that much, so I'll just change it to the detupled version.